### PR TITLE
ref(dropdownButton): Left justify text, right justify chevron

### PR DESCRIPTION
**Pull request by vuluongj20**
Establish a UI pattern on `DropdownButton`s: left-justified text and right-justified chevrons
image

This is a general pattern that we'd want to use on all dropdown buttons. There's no visual diff that I'm aware of — existing components use custom CSS to implement the same layout (which we can now delete 🎉).

### CODE REVIEW

File: static/app/components/dropdownButton.tsx [hunk 1]
```diff
@@ -46,14 +46,19 @@ function DropdownButton({
...
 
-const StyledChevron = styled(IconChevron)`
-  margin-left: ${space(0.75)};
+const ChevronWrap = styled('div')`
+  display: flex;
+  align-items: center;
+  margin-left: auto;
+  padding-left: ${space(0.75)};
   flex-shrink: 0;
 `;
```
Comment (vuluongj20) [inline static/app/components/dropdownButton.tsx:63]: This is the main code change. There's now a `ChevronWrap` with a `margin-left: auto` to push to to the right side.

**Review by evanpurkhiser — APPROVED** [review]

File: static/app/components/button.tsx [hunk 2]
```diff
@@ -475,6 +475,7 @@ const Icon = styled('span')<IconProps & Omit<StyledButtonProps, 'theme'>>`
   display: flex;
   align-items: center;
   margin-right: ${getIconMargin};
+  flex-shrink: 0;
```
Comment (vuluongj20) [inline static/app/components/button.tsx:478]: Making sure that icons don't get squished when the button shrinks.

File: static/app/components/environmentPageFilter.tsx [hunk 3]
```diff
@@ -117,20 +115,8 @@ function EnvironmentPageFilter({
...
 `;
 
-const DropdownTitle = styled('div')`
-  display: flex;
-  align-items: center;
-  flex: 1;
-  width: max-content;
-  min-width: 0;
-`;
-
```
Comment (vuluongj20) [inline static/app/components/environmentPageFilter.tsx:135]: Yayyyy

Comment (evanpurkhiser) [reply static/app/components/environmentPageFilter.tsx:135]: LOVE

**Pull request merged**